numpy/core/tests/test_deprecations.py

```diff
@@ -728,3 +732,14 @@ def test_deprecated(self, arr, exc_type):
 
         out = np.lib.function_base._trim_zeros_old(arr)
         assert_array_equal(arr, out)
+
+    def test_type_error(self):
+        arr = np.array(["0", "a", "b", ""])
+        with pytest.warns(FutureWarning), pytest.raises(TypeError):
+            np.lib.function_base._trim_zeros_new(arr)
+
+    def test_warning(self):
+        arr = [0, 0, [1], 0]
+        with pytest.warns(np.VisibleDeprecationWarning):
+            out = np.lib.function_base._trim_zeros_new(arr)
```

Comment:
We could be testing the public API here, right?
```suggestion
            out = np.lib.function_base.trim_zeros(arr)
```

Reply:
You're right, yes; changed in https://github.com/numpy/numpy/pull/17131/commits/52b2f6bc1209000be372987a07f0b079e06bb7c0.